### PR TITLE
Add Google Analytics

### PR DESCRIPTION
**Pull request by nickserv**

### CODE REVIEW

**Comment by BTMPL** [comment]
Please keep in mind that GA requires an explicit opt-in GDPR compliant implementation.

**Comment by nickserv** [comment]
We're using this to anonymize the IP, is it enough? https://support.google.com/analytics/answer/2763052

**Comment by BTMPL** [comment]
> We're using this to anonymize the IP, is it enough? https://support.google.com/analytics/answer/2763052

I think you also at least need to disable the usage of collected data for ad purposes - https://www.blastanalytics.com/blog/gdpr-need-consent-for-google-analytics-tracking

image

IANAL, but this should then be good enough to avoid any actions. We can keep this discussion as proof we're not gathering / sharing such data.

**Comment by nickserv** [comment]
We don't care about ad tracking, so I've disabled `allowAdFeatures`, which should also disable everything in the data collection options.

**Review by BTMPL — APPROVED** [review]

**Comment by nickserv** [comment]
The build seems to still be randomly be failing on sharp. It always works after clearing the cache though. I'm not sure if this needs to be fixed before merging.

**Comment by vcarl** [comment]
I contacted Netlify support about that because I've never encountered it before, they basically said 🤷 Directed me to https://community.netlify.com/t/support-guide-my-site-deploy-fails-unless-netlifys-build-cache-is-cleared/3035 but none of those appear to apply.

File: yarn.lock [hunk 1]
```diff
@@ -729,6 +729,13 @@
   dependencies:
     regenerator-runtime "^0.13.2"
 
+"@babel/runtime@^7.10.2":
```
Comment (vcarl) [inline yarn.lock:732]: Lockfile is up to date? I'm surprised to see this change

**Review by vcarl — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (nickserv) [reply yarn.lock:732]: It's a dependency of `gatsby-plugin-google-analytics@^2.3.5`

**Comment by nickserv** [comment]
@vcarl That's frustrating. I suppose it could be the sharp binary, but I doubt Node is updating every single build. Should we just merge for now and see what happens?

**Comment by vcarl** [comment]
Right, it seems cache related more than anything else to me. I'm down to merge, worst case we have to manually retrigger

File: yarn.lock (hunk 1 above)
Comment (vcarl) [reply yarn.lock:732]: Dang that's annoying. This is why I don't like plugins haha

**Comment by nickserv** [comment]
Got it to work!

**Comment by bpas247** [comment]
> The build seems to still be randomly be failing on sharp. It always works after clearing the cache though. I'm not sure if this needs to be fixed before merging.

I did find [this](https://github.com/gatsbyjs/gatsby/issues/4693#issuecomment-526404864), although I'm not sure if we want to manually install sharp just for Netlify's sake 😛

**Comment by nickserv** [comment]
I don't know how we'd want to integrate this into our build process either. Disabling postbuild for all packages seems a bit drastic. Would be nice if we could somehow selectively build just sharp with the different options.

**Pull request merged**